infrastructure/network/randomaddress/address_randomize.go

```diff
@@ -0,0 +1,43 @@
+package randomaddress
```

Comment:
I might be missing something, but I don't understand why do we need a separate package for RandomAddress?

Note that later down the road, most probably the address selection algorithm will get more complex, including managing multiple address lists within the address manager (similar or not to the New/Tried distinction + buckets in the legacy address manager).

Therefore a generic object that gets an address list and returns a random address out of it will hurt our extensibility rather then helping it.